tap_chargebee/streams/base.py

```diff
@@ -71,11 +71,42 @@ def generate_catalog(self):
             'metadata': singer.metadata.to_list(mdata)
         }]
 
+    def appendCustomFields(self, record):
+        listOfCustomFieldObj = ['addon', 'plan', 'subscription', 'customer']
+        custom_fields = {}
+        event_custom_fields = {}
+        if self.ENTITY == 'event':
+            content = record['content']
+            words = record['event_type'].split("_")
+            counter = 1;
+            content_obj = ""
+            for word in words:
+                if counter != len(words):
+                    content_obj = str(content_obj) + word 
+                    if(counter + 1 != len(words)):
+                        content_obj = content_obj + "_"
+                counter = counter + 1
```

Comment:
Can this be made more clear with a unit test? This seems like it's either doing nothing or rebuilding the original `event_type` string, with the last 1 or 2 sections removed. A string split, slice, and join would be more clear in the latter case.

Reply:
From events_type( a field in Event stream's record) we are fetching the main object for which, this event triggered (sample value "credit_note_updated" ==> "credit_note").  Full resource of that object will be present in the event object. The resource may have the custom fields and we are constructing all the custom fields as a single JSON field.

Changed the code as you mentioned.